### DISCUSSION TITLE
Zmk not waking up from deep sleep

### DISCUSSION THREAD

**Discussion by matdurand** [christianselig/caldera-keyboard#4] [Q&A] [answered]
Hey, I built a Caldera, and I have an issue with deep sleep. After it goes into deep sleep, the only way to wake the keyboard is to hit the reset switch. None of the keys are able to wake it. Is this normal behavior for ZMK? I would assume there would a delay for bluetooth to reconnect, but that any of the keys would wake it up. I couldn't find anything on the ZMK doc on this.

It's kind of annoying to have to press a tiny reset switch, and also I have to press both side at the same time otherwise the right side won't wake and sync with the left.

**Comment by christianselig** [comment]
Huh, I don't have that problem, if I tap a key on each side of the keyboard it wakes up, I haven't hit the reset button in months. ZMK admittedly has quite a few settings that I don't totally understand, I'd poke my head in the Discord if I were you and see if they have any suggestions, I'm sure it's just a config entry somewhere

**Reply by matdurand** [reply, marked as answer]
I found some interesting stuff just browsing around the discord. I'm trying right now but my guess is that you're missing `wake-source;` here: https://github.com/christianselig/caldera-keyboard/blob/main/zmk-config-caldera/config/boards/shields/caldera/caldera.dtsi#L31. I need to wait for the keyboard to go to sleep again to confirm.

I'm assuming the zmk config in the repo is not your actual config since I seem to remember you mentioned in the video you're using colemak layout, so maybe you had the wake-source in your original and not in the official repo?

**Reply by matdurand** [reply]
Yup, it works now! Do you want a PR or are you gonna fix it yourself since it's a single line?
https://github.com/matdurand/zmk-config-caldera/blob/main/config/boards/shields/caldera/caldera.dtsi#L31

**Reply by christianselig** [reply]
A PR would be great, that way you can get the credit for finding it haha

**Reply by matdurand** [reply]
PR created. Thanks for all this work you did!

**Reply by matdurand** [reply]
and I posted a picture of my board here: https://makerworld.com/en/models/545431#profileId-463245
I love it so far!
